app/views/unattended/provisioning_templates/snippet/puppet.conf.erb

```diff
@@ -41,9 +41,9 @@ ssldir = <%= ssl_dir %>
 [agent]
 pluginsync      = true
 report          = true
-<%- if @host.puppet_ca_server.strip -%>
-ca_server       = <%= @host.puppet_ca_server %>
+<%- if host_puppet_ca_server.present? -%>
+ca_server       = <%= host_puppet_ca_server %>
 <%- end -%>
 certname        = <%= @host.certname %>
-environment     = <%= @host.environment %>
-server          = <%= @host.puppetmaster %>
+environment     = <%= host_puppet_environment %>
```

Comment:
I guess this is now also optional. Perhaps this should also be made optional, just like the CA server?